nexus/src/app/instance.rs

```diff
@@ -1362,6 +1361,43 @@ impl super::Nexus {
         }
     }
 
+    async fn propolis_addr_for_instance(
+        &self,
+        instance_lookup: &lookup::Instance<'_>,
+        action: authz::Action,
+    ) -> Result<SocketAddr, Error> {
+        let (.., authz_instance, instance) =
+            instance_lookup.fetch_for(action).await?;
+        match instance.runtime_state.state.0 {
+            InstanceState::Running
+            | InstanceState::Rebooting
+            | InstanceState::Migrating
+            | InstanceState::Repairing => {
+                let ip_addr = instance
+                    .runtime_state
+                    .propolis_ip
+                    .ok_or_else(|| {
+                        Error::internal_error(
+                            "instance's hypervisor IP address not found",
+                        )
+                    })?
+                    .ip();
+                Ok(SocketAddr::new(ip_addr, PROPOLIS_PORT))
```

Comment:
I think this might not be the proper port?

Reply:
Ah good, thanks.  I was not sure if we required a different port here for the serial console stuff.